src/js/form-model.js

```diff
@@ -532,9 +532,13 @@ FormModel.prototype.setInstanceIdAndDeprecatedId = function() {
         // set the instanceID value to empty
         instanceIdEl.textContent = '';
 
+        const namespace = instanceIdEl.namespaceURI;
+
         // add deprecatedID node if necessary
         if ( !deprecatedIdEl ) {
-            deprecatedIdEl = parser.parseFromString( '<deprecatedID/>', 'text/xml' ).documentElement;
+            const nsPrefix = namespace ? this.getNamespacePrefix( namespace ) : '';
+            const nsDeclaration = namespace ? `xmlns:${nsPrefix}="${namespace}"` : '';
+            deprecatedIdEl = parser.parseFromString( `<${nsPrefix ? nsPrefix + ':' : ''}deprecatedID ${nsDeclaration}/>`, 'text/xml' ).documentElement;
```

Comment:
String concatenation of this always raises my eyebrows a bit. It's probably reasonable enough here, but worth noting for posterity that this is a small risk in cases where special characters are used. Probably more likely for the namespace URI than the prefix, although I was able to cause test failures with both.

Reply:
Good point. We could create elements and attributes here using createElement etc. Laziness on my part! :)